src/components/SignUpForm/SignUpForm.js

```diff
@@ -0,0 +1,113 @@
+import { collection,addDoc,setDoc,doc } from "@firebase/firestore";
+import Link from "next/link";
+import { db, auth} from "@/firebase/config";
+
+
+const SignUpForm = () => {
+    const [email, setEmail] = useState(null);
+    const [password, setPassword] = useState(null);
+    const [confirmPassword, setConfirmPassword] = useState(null);
+    const [name,setName] =useState(null)
+    const [error, setError] = useState(" ");
+    const router = useRouter();
+
+   
+    const submitHandler = async () =>{
+        /*const adding = await setDoc(doc(db, "users",122), 
+            userDetails
+        );*/
+        
+        if(confirmPassword != password){
+            setError("Error: (Passwords do not match)");
+        }else if(password.length <6){
+            setError("Error: (Weak password)");
+        }else{
+            if(auth){
+                signOut(auth);
+            }
+            console.log("auth: ",auth);
+            try{
+                const userData = await createUserWithEmailAndPassword(auth, email,password);
+                await updateProfile(auth.currentUser,{displayName:`${name}`});
+                router.push("/");
+            }catch(error){
+    
+                const errorCode = error.code;
+                const errorMessage = error.message;
+                console.log("error code",errorCode);
+                console.log("error message: ",errorMessage);
+                setError(errorMessage);
+            }
+        }
+    }
+
+    
+    return (
+        <div className={styles.container}>
+            <div>
+                <div className={styles.greeting}>Get started</div>
+                <div className={styles.subGreeting}>Provide the details below to create an account</div>
+                <div className={styles.error}>{error}</div>
+                <div className={styles.inputLabel}>Name</div>
+                <input
+                    className={styles.inputField}
+                    type="text"
+                    placeholder="Enter your full name"
+                    value={name}
+                    required
+                    onChange={(e) => setName(e.target.value)}
+                />
+                <div className={styles.inputLabel}>Email</div>
+                <input
+                    className={styles.inputField}
+                    type="text"
+                    placeholder="Enter your email"
+                    value={email}
+                    required
+                    onChange={(e) => setEmail(e.target.value)}
+                />
+                {/*<div className={styles.inputLabel}>Account type</div>
+                <select onChange={(e) => setUserType(e.target.value)} className={styles.inputField}>
+                    <option value="Ordinary User">Ordinary User</option>
+                    <option value="Super User">Super User</option>
+                </select>*/}
```

Comment:
Remove some of the extra comments, unless we need it.

Reply:
okay, I will try to fix that latetr

Reply:
I will fix that right now